### PR TITLE
Add test cases for sorting based on length and character value in flux

### PR DESCRIPTION
**Pull request by adamperlin**
Adds test cases to address #224. Results: strings are sorted by first character values initially and then length secondarily if string prefixes are the same.

### CODE REVIEW

File: query/querytest/test_cases/string_sort_length.flux [hunk 1]
```diff
@@ -0,0 +1,3 @@
+from(db: "test")
```
Comment (aanthony1243) [inline query/querytest/test_cases/string_sort_length.flux]: This test case isn't intrinsically different from the first one, it's still doing a dictionary sort.   See my final review comment for how to proceed.

**Comment by aanthony1243** [comment]
also, you are off to a great start.  This language is best learned by just doing things with it and finding out what happens.  keep it up!

**Comment by adamperlin** [comment]
I added the two tests as requested. Testing `max()` on strings was a good idea; it revealed a panic. 
`error: panic: interface conversion: interface is nil, not execute.DoStringRowSelector
goroutine 21 [running]:
runtime/debug.Stack(0x14bba60, 0x14b9b40, 0xc42034cb80)
	/usr/local/Cellar/go/1.10.3/libexec/src/runtime/debug/stack.go:24 +0xa7
github.com/influxdata/platform/query/execute.(*poolDispatcher).Start.func1.1(0xc420520a50)
	/Users/adamperlin/go/src/github.com/influxdata/platform/query/execute/dispatcher.go:67 +0xb9
panic(0x14b9b40, 0xc42034cb80)
	/usr/local/Cellar/go/1.10.3/libexec/src/runtime/panic.go:502 +0x229
github.com/influxdata/platform/query/execute.(*rowSelectorTransformation).Process.func1(0x15bff40, 0xc420218050, 0x1543801, 0xc42034cb40)
	/Users/adamperlin/go/src/github.com/influxdata/platform/query/execute/selector.go:183 +0x258
github.com/influxdata/platform/query/execute.(*ColListBlock).Do(0xc420218050, 0xc42034cb40, 0x0, 0x0)
	/Users/adamperlin/go/src/github.com/influxdata/platform/query/execute/block.go:602 +0x3c
github.com/influxdata/platform/query/execute.(*rowSelectorTransformation).Process(0xc4200423c0, 0x255fce4fb6a41b39, 0x928afa2d6a496c96, 0x15bc6c0, 0xc420218050, 0x1, 0x2)
	/Users/adamperlin/go/src/github.com/influxdata/platform/query/execute/selector.go:172 +0x26b
github.com/influxdata/platform/query/execute.processMessage(0x15bc800, 0xc4200423c0, 0x15b72c0, 0xc4201be260, 0x3, 0x0, 0x0)
	/Users/adamperlin/go/src/github.com/influxdata/platform/query/execute/transport.go:199 +0x200
github.com/influxdata/platform/query/execute.(*consecutiveTransport).processMessages(0xc420221500, 0xa)
	/Users/adamperlin/go/src/github.com/influxdata/platform/query/execute/transport.go:156 +0xa1
github.com/influxdata/platform/query/execute.(*consecutiveTransport).(github.com/influxdata/platform/query/execute.processMessages)-fm(0xa)
	/Users/adamperlin/go/src/github.com/influxdata/platform/query/execute/transport.go:137 +0x34
github.com/influxdata/platform/query/execute.(*poolDispatcher).run(0xc420520a50, 0x15bb280, 0xc420412b70)
	/Users/adamperlin/go/src/github.com/influxdata/platform/query/execute/dispatcher.go:116 +0x4b
github.com/influxdata/platform/query/execute.(*poolDispatcher).Start.func1(0xc420520a50, 0x15bb280, 0xc420412b70)
	/Users/adamperlin/go/src/github.com/influxdata/platform/query/execute/dispatcher.go:70 +0x95
created by github.com/influxdata/platform/query/execute.(*poolDispatcher).Start
	/Users/adamperlin/go/src/github.com/influxdata/platform/query/execute/dispatcher.go:55 +0x7e`

I tracked down the source: https://github.com/influxdata/platform/blob/master/query/functions/max.go#L117
There's currently no `max` implementation for strings. I'll open a new issue.

File: query/querytest/query_test.go [hunk 2]
```diff
@@ -21,6 +21,7 @@ var skipTests = map[string]string{
 	"group_ungroup":             "influxql/flux disagreement on keycols (https://github.com/influxdata/platform/issues/95)",
 	"window":                    "ordering of results differs between queries (https://github.com/influxdata/platform/issues/96)",
 	"window_group_mean_ungroup": "error in influxql: failed to run query: timeValue column \"_start\" does not exist (https://github.com/influxdata/platform/issues/97)",
+	"string_max":                "panic because no max() implementation for string",
```
Comment (aanthony1243) [inline query/querytest/query_test.go]: need to reference an issue URL here.

Comment (adamperlin) [reply query/querytest/query_test.go]: I left a reference to the new panic issue.

**Pull request merged**